### PR TITLE
DevKit updates

### PR DESCRIPTION
**Pull request by ElectricMaxxx**
So this is the PR on master, the one on 2.0 will be closed except the stricter style checks and the testing script discussion this looks ready now.

### CODE REVIEW

**Comment by ElectricMaxxx** [comment]
for the stricter style rules: i like them :-)

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+-->
+
+<!--
+    If you are reporting a bug, please try to fill in the following.
+    Otherwise remove it.
+-->
+
+### Environment
+
+#### Symfony paackages
```
Comment (dbu) [inline .github/ISSUE_TEMPLATE.md]: typo: paackages

File: .styleci.yml [hunk 2]
```diff
@@ -1,7 +1,31 @@
...
 enabled:
-  - ordered_use
+  - combine_consecutive_unsets
   - short_array_syntax
-  
+  - newline_after_open_tag
+  - no_php4_constructor
+  - no_useless_else
+  - ordered_use
+# Comment strict rules for the moment. Should be uncomment later to see StyleCI PR results
```
Comment (dbu) [inline .styleci.yml]: erm, seems things are not commented out below. is this comment still relevant?

File: .travis.yml [hunk 3]
```diff
@@ -22,10 +37,11 @@ matrix:
       env: DEPS=dev SYMFONY_VERSION=3.3.*
     - php: 5.6
       env: COMPOSER_FLAGS="--prefer-lowest" SYMFONY_VERSION=2.8.* SYMFONY_DEPRECATIONS_HELPER=weak
-    - php: 7.1
-      env: SYMFONY_VERSION=3.1.*
-    - php: hhvm
```
Comment (dbu) [inline .travis.yml]: we might need to update readme or other places if we drop hhvm support

File: .travis.yml [hunk 4]
```diff
@@ -11,35 +26,47 @@ cache:
...
+    - php: 7.1
+      env: SYMFONY_VERSION=3.2.*
+
+    - env: TEST_INSTALLATION=true
   fast_finish: true
+  allowed_failures:
+    - php: 7.1
+      env: SYMFONY_VERSION=3.4.*
+    - env: TEST_INSTALLATION=true
+
```
Comment (ElectricMaxxx) [inline .travis.yml]: @dbu can you give me a hand with the `allowed_failures` here? I know we do not like it. but i would like to see future states here, but i it does not really work the way i do it: https://travis-ci.org/symfony-cmf/seo-bundle

Comment (ElectricMaxxx) [reply .travis.yml]: ups, i should write it the right way

**Comment by ElectricMaxxx** [comment]
I tried everything now here:
* using phpunit 6.0 only -> had to increase Symfony version on 3.2+ -> our future major release
* using several versions of `matthiasnoback/*`, but there it they differ only on `^5.0` or `^6.0` of phpunit, so there is no middle

i will try to get an explicit newer version of `sebastian/exporter` now.

File: .travis.yml [hunk 5]
```diff
@@ -11,35 +26,46 @@ cache:
...
   - composer self-update
   - if [ "$DEPS" = "dev" ]; then perl -pi -e 's/^}$/,"minimum-stability":"dev"}/' composer.json; fi
   - if [ "$SYMFONY_VERSION" != "" ]; then composer require symfony/symfony:${SYMFONY_VERSION} --no-update; fi
+  - export BRANCH=$(if [ "$TRAVIS_PULL_REQUEST" == "false" ]; then echo $TRAVIS_BRANCH; else echo $TRAVIS_PULL_REQUEST_BRANCH; fi)
 
 install: travis_wait composer update --prefer-dist $COMPOSER_FLAGS
 
-before_script: vendor/symfony-cmf/testing/bin/travis/phpcr_odm_doctrine_dbal.sh
-
-script: vendor/bin/simple-phpunit
```
Comment (dbu) [inline .travis.yml]: does make test really run vendor/bin/simple-phpunit or could it be that it runs something else?

Comment (ElectricMaxxx) [reply .travis.yml]: when debbung into it: yes.

**Comment by ElectricMaxxx** [comment]
uhh, by requiring `sebastion/exporter` on my own it works, but got an other issue now :-)

File: composer.json [hunk 6]
```diff
@@ -12,21 +12,24 @@
...
-        "sonata-project/seo-bundle": "^1.1.3|^2.0",
-        "doctrine/collections": "^1.0"
+        "sonata-project/seo-bundle": "^2.2",
+        "doctrine/collections": "^1.0",
+        "symfony/symfony": "^2.8",
+        "phpcr/phpcr-utils": "^1.2.9"
     },
     "require-dev": {
-        "symfony-cmf/testing": "^2.0",
-        "doctrine/phpcr-odm": "^1.4.2",
```
Comment (dbu) [inline composer.json]: we now install a too old version of phpcr-odm it seems

Comment (ElectricMaxxx) [reply composer.json]: oh second wasn't wanted

**Comment by dbu** [comment]
lets add a conflicts for phpcr-utils < 1.3 :-(

File: composer.json [hunk 7]
```diff
@@ -12,21 +12,24 @@
...
     "require": {
-        "php": "^5.6|^7.0",
+        "php": "^7.1",
         "symfony/framework-bundle": "^2.8|^3.0",
-        "sonata-project/seo-bundle": "^1.1.3|^2.0",
-        "doctrine/collections": "^1.0"
+        "sonata-project/seo-bundle": "^2.2",
+        "doctrine/collections": "^1.0",
+        "symfony/symfony": "^2.8",
+        "phpcr/phpcr-utils": "^1.2.9"
```
Comment (dbu) [inline composer.json]: did we start depending on this? does this make sense? do we not either depend on phpcr-odm or on nothing phpcr? if this really belongs here, then we don't need the conflicts section and can just do `^1.3.0` here.

File: composer.json [hunk 8]
```diff
@@ -51,5 +54,10 @@
         "branch-alias": {
             "dev-master": "2.1-dev"
         }
-    }
+    },
+    "conflicts":{
```
Comment (dbu) [inline composer.json]: correct spelling would be `conflict` without the s at the end: https://getcomposer.org/doc/04-schema.md#conflict

unfortunately composer does not complain when it encounters unknown keys in the file.

File: composer.json (hunk 7 above)
Comment (ElectricMaxxx) [reply composer.json]: we didn't had it here, i put it there to get a better version, so i will clean it up, i even pushed the requirement on symfony.

Comment (ElectricMaxxx) [reply composer.json]: I even moved sebastian/exporter to conflict

**Comment by dbu** [comment]
yeehaa! its green!
the stand alone installation test fails, looks like a mixup between symfony versions in the commands we run.

**Comment by ElectricMaxxx** [comment]
yea, but this one should fail until sonata-seo is released.

**Comment by dbu** [comment]
we can address the stand alone build in a separate PR. unless you are aware of issues with this, lets squash and merge!

**Comment by ElectricMaxxx** [comment]
btw: i see it green on one of the travis builds.

**Pull request merged**